service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java

```diff
@@ -142,6 +144,24 @@ private void retry(RequestContext requestContext, Handler<RestResponse> response
               return;
             }
 
+            // no need to generate warn log when schema or service not exist
+            if (HttpStatusClass.CLIENT_ERROR.equals(HttpStatusClass.valueOf(response.statusCode()))) {
+              try {
+                if (JsonUtils.readValue(bodyBuffer.getBytes(), Map.class).containsKey("errorCode")) {
+                  @SuppressWarnings("unchecked")
```

Comment:
@SuppressWarnings("unchecked")  should also cover the above line.   
```
JsonUtils.readValue(bodyBuffer.getBytes(), Map.class)
```
is a big operation, it's better only call it once.